### PR TITLE
Enable TypeScript strict mode

### PR DESCRIPTION
**Pull request by johnnyomair**
---

## PR Checklist

-   [x] Link to the respective task if one exists: [COM-243](https://vivid-planet.atlassian.net/browse/COM-243)

[COM-243]: https://vivid-planet.atlassian.net/browse/COM-243?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: api/tsconfig.json [hunk 1]
```diff
@@ -14,7 +14,6 @@
         "esModuleInterop": true,
         "strict": true,
         "strictPropertyInitialization": false,
```
Comment (johnnyomair) [inline api/tsconfig.json:16]: If we remove this line, we need to add the "definite assignment assertion" (!) operator to all entities and blocks. Do we want to change that?

File: admin/src/common/masterMenuData.tsx [hunk 2]
```diff
@@ -21,7 +22,8 @@ export const pageTreeCategories: AllCategories = [
     },
 ];
 
-export const pageTreeDocumentTypes = {
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export const pageTreeDocumentTypes: Record<string, DocumentInterface<any, any>> = {
```
Comment (johnnyomair) [inline admin/src/common/masterMenuData.tsx:26]: Without this explicit type annotation, we run into lot's of errors like:

```
Type '{ Page: DocumentInterface<Pick<GQLPage, "content" | "seo">, GQLPageInput> & DependencyInterface; Link: DocumentInterface<...> & DependencyInterface; }' is not assignable to type 'Record<string, DocumentInterface<Record<string, unknown>, Record<string, unknown>>> | ((category: string) => Record<string, DocumentInterface<Record<string, unknown>, Record<...>>>)'.
  Type '{ Page: DocumentInterface<Pick<GQLPage, "content" | "seo">, GQLPageInput> & DependencyInterface; Link: DocumentInterface<...> & DependencyInterface; }' is not assignable to type 'Record<string, DocumentInterface<Record<string, unknown>, Record<string, unknown>>>'.
    Property 'Page' is incompatible with index signature.
      Type 'DocumentInterface<Pick<GQLPage, "content" | "seo">, GQLPageInput> & DependencyInterface' is not assignable to type 'DocumentInterface<Record<string, unknown>, Record<string, unknown>>'.
        Types of property 'updateMutation' are incompatible.
          Type 'TypedDocumentNode<GQLUpdatePageMutation, GQLUpdatePageMutationVariables<Pick<GQLPage, "content" | "seo">>> | undefined' is not assignable to type 'TypedDocumentNode<GQLUpdatePageMutation, GQLUpdatePageMutationVariables<Record<string, unknown>>> | undefined'.
            Type 'TypedDocumentNode<GQLUpdatePageMutation, GQLUpdatePageMutationVariables<Pick<GQLPage, "content" | "seo">>>' is not assignable to type 'TypedDocumentNode<GQLUpdatePageMutation, GQLUpdatePageMutationVariables<Record<string, unknown>>>'.
              Type 'GQLUpdatePageMutationVariables<Record<string, unknown>>' is not assignable to type 'GQLUpdatePageMutationVariables<Pick<GQLPage, "content" | "seo">>'.
                Type 'Record<string, unknown>' is missing the following properties from type 'Pick<GQLPage, "content" | "seo">': content, seo
```

As for the `any`s: We should probably change this in the library to not use `unknown`, but `any`

File: admin/src/App.tsx [hunk 3]
```diff
@@ -76,7 +76,7 @@ export function App() {
                             }}
                         >
                             <IntlProvider locale="en" messages={getMessages()}>
-                                <LocaleProvider resolveLocaleForScope={(scope: ContentScope) => scope.domain}>
+                                <LocaleProvider resolveLocaleForScope={(scope) => scope.domain}>
```
Comment (johnnyomair) [inline admin/src/App.tsx:79]: We'd need module augmentation for this to work correctly

**Comment by johnnyomair** [comment]
I could also evaluate and try to fix this in Demo, what do you prefer?

File: admin/src/App.tsx (hunk 3 above)
Comment (nsams) [reply admin/src/App.tsx:79]: how does it work now? is scope with this change any?

Comment (johnnyomair) [reply admin/src/App.tsx:79]: It's ContentScopeInterface, which basically is `Record<string, any>`

Comment (nsams) [reply admin/src/App.tsx:79]: can we add module augmentation? (in a future PR)

File: api/tsconfig.json (hunk 1 above)
Comment (nsams) [reply api/tsconfig.json:16]: that is something out of scope for this PR

File: admin/src/App.tsx (hunk 3 above)
Comment (johnnyomair) [reply admin/src/App.tsx:79]: > can we add module augmentation? (in a future PR)

Probably. But it needs to be done in the library

**Comment by nsams** [comment]
> I could also evaluate and try to fix this in Demo, what do you prefer?

we can start new projects with strict mode, and enable it also in demo. The order doesn't matter.

**Review by nsams — DISMISSED** [review]

**Review by dkarnutsch — APPROVED** [review]

**Review by nsams — APPROVED** [review]

**Comment by johnnyomair** [comment]
@dkarnutsch @nsams I've updated this PR, please review once more, thanks!

File: admin/src/common/masterMenuData.tsx (hunk 2 above)
Comment (johnnyomair) [reply admin/src/common/masterMenuData.tsx]: Created a task for this: https://vivid-planet.atlassian.net/browse/COM-1077

**Review by nsams — APPROVED** [review]

**Pull request merged**